File: arkhe_multilang/rosetta/Agent.cs
using System;

class Agent {
    public string Name { get; set; }
    public int Counter { get; set; }
    public Agent(string name) {
        Name = name;
        Counter = 0;
    }
}

class Program {
    static void SendMessage(Agent sender, Agent receiver) {
        receiver.Counter++;
        Console.WriteLine($"{sender.Name} -> {receiver.Name}: {receiver.Counter}");
    }

    static void Main() {
        Agent alice = new Agent("Alice");
        Agent bob = new Agent("Bob");
        SendMessage(alice, bob);
        SendMessage(alice, bob);
    }
}

File: arkhe_multilang/rosetta/Agent.java
public class Agent {
    String name;
    int counter;

    public Agent(String name) {
        this.name = name;
        this.counter = 0;
    }

    public static void sendMessage(Agent sender, Agent receiver) {
        receiver.counter++;
        System.out.println(sender.name + " -> " + receiver.name + ": " + receiver.counter);
    }

    public static void main(String[] args) {
        Agent alice = new Agent("Alice");
        Agent bob = new Agent("Bob");
        sendMessage(alice, bob);
        sendMessage(alice, bob);
    }
}
